### ISSUE TITLE
Add NumDescendants() to HoeffdingTree

### ISSUE THREAD

**Issue by rcurtin** [mlpack/mlpack#977]
This comes from a conversation in IRC:

```
19:53 < cult-> what methods are available to measure how big is the Hoeffding tree? i want to somehow verify its robustness before i apply Classify().
19:53 < cult-> i can see the methods, like NumChildren() MinSamples(), MaxSamples() but i don't know which one to use
19:54 < cult-> what is MajorityClass()?
19:54 < cult-> or how should I approach this problem? can I measure on how many observations the tree has been already trained? or how many times?
19:54 < cult-> s/measure/request
19:57 < rcurtin> so you could use NumChildren() recursively but that might be a bit irritating
19:57 < rcurtin> here is a simple code snippet, I think it will work:
19:57 < rcurtin> std::stack<HoeffdingTree<>*> stack;
19:57 < rcurtin> stack.push(&tree); // 'tree' is the root of the tree
19:57 < rcurtin> size_t nodes = 0;
19:57 < rcurtin> while (!stack.empty())
19:57 < rcurtin> {
19:57 < rcurtin>   HoeffdingTree<>* node = stack.top();
19:57 < rcurtin>   stack.pop();
19:57 < rcurtin>   nodes += node.NumChildren();
19:57 < rcurtin>   for (size_t i = 0; i < node.NumChildren(); ++i)
19:58 < rcurtin>     stack.push(&node.Child(i));
19:58 < rcurtin> }
19:58 < rcurtin> MajorityClass() is the class most often seen by that particular node during training
19:59 < rcurtin> so if you asked for the predicted class for some sample, the tree would percolate that sample to the appropriate leaf node and then the predicted class would be MajorityClass() in that leaf 
                 node
19:59 < rcurtin> I should add a NumDescendants() function to HoeffdingTree<>, let me open an issue for that since I am doing other things right now
```

The code that I wrote will probably work but tests will also needed to be added.

**Comment by s1998** [comment]
@rcurtin  Can I work on this ?

**Comment by rcurtin** [comment]
Sure, should be an easy addition. The code I gave should work, just be sure to write some simple test(s) for it.

**Comment by rcurtin** [comment]
Fixed via #1006.
